src/optimization.py

```diff
@@ -5,41 +5,59 @@
 
 
 class Parameters:
+    """
+    A class to hold the parameters of the ev charging station optimization.
+
+    :param z0: ...
+    :param v0: ...
+    :param Ts: Time of use cost array
+    :param base_tariff_overstay: ...
+    :param eff: charger efficiency
+    :param dcm_choices: list of possible choices for the discrete choice model
+    :param soft_v_eta: softening equality constraint for v; to avoid numerical error
+    :param opt_eps: ...
+    """
 
-    def __init__(self, 
-                z0 = np.array([1,1,1,1]).reshape(4,1) ,
-                v0 = np.array([0.333, 0.333, 0.333]).reshape(3,1) ,  
-                Ts = 0.25, 
-                base_tarriff_overstay = 1.0, 
-                eff = 1, 
-                soft_v_eta = 1e-3, 
-                opt_eps = 0.0001, 
-                TOU = np.ones((96,))):
-        
+    def __init__(self,
+                 z0=np.array([1, 1, 1, 1]).reshape(4, 1),
+                 v0=np.array([0.333, 0.333, 0.333]).reshape(3, 1),
+                 Ts=0.25,
+                 base_tariff_overstay=1.0,
+                 TOU=np.ones((96,)),
+                 eff=1,
+                 dcm_choices=['charging with flexibility', 'charging asap', 'leaving without charging'],
+                 soft_v_eta=1e-3,
+                 opt_eps=0.0001,
+                 ):
         self.v0 = v0
         self.z0 = z0
         self.Ts = Ts
-        self.base_tariff_overstay = base_tarriff_overstay
+        self.base_tariff_overstay = base_tariff_overstay
         self.TOU = TOU
-        self.eff = eff # power efficiency
-        self.dcm_choices = ['charging with flexibility', 'charging asap', 'leaving without charging']
-        self.soft_v_eta = soft_v_eta #softening equality constraint for v; to avoid numerical error
+        self.eff = eff
+        self.dcm_choices = dcm_choices
+        self.soft_v_eta = soft_v_eta
         self.opt_eps = opt_eps
-        
-        assert len(self.TOU) == int(24 / self.Ts), "Mismatch between TOU cost array size and discretization steps"
 
+        assert len(self.TOU) == int(24 / self.Ts), "Mismatch between TOU cost array size and discretization steps"
 
 
 class Problem:
     """
-    time, int, user interval 
-    duration, int, number of charging intervals 
+    A class containing variables for the charging optimization problem.
+
+    :param xxx: ...
```

Comment:
*:param par:*
*:param event: Key information that station need when charging and optimizing.*
*:param user_time: User arrival time*
*:param e_need: Electricity user need to charge*
*:param user_duration: Duration of user staying*
*:param user_overstay_duration: Duration of user overstaying*
*:param station_pow_max: Station maximum power*
*:param station_pow_min: Station minimum power*
*:param power_rate: Charging power rate*
*:param dcm_charging_flex_params: FLEX charging parameter of DCM*
*:param dcm_charging_asap_params: ASAP charging parameter of DCM*
*:param dcm_leaving_params:Leaving without charging parameter of DCM*
*:param THETA:*
*:param N_flex:*
*:param N_asap:*